test/e2e/terminal.test.ts

```diff
@@ -0,0 +1,77 @@
+import { test, expect } from "@playwright/test"
+import * as fs from "fs"
+import { tmpdir } from "os"
+import * as path from "path"
+import util from "util"
+import * as cp from "child_process"
+import { STORAGE } from "../utils/constants"
+import { CodeServer } from "./models/CodeServer"
+
+test.describe("Integrated Terminal", () => {
+  // Create a new context with the saved storage state
+  // so we don't have to logged in
+  const options: any = {}
+  const testFileName = "test.txt"
+  const testString = "new string test from e2e test"
+  let codeServer: CodeServer
+  let tmpFolderPath: string = ""
+  let tmpFile: string = ""
+
+  // TODO@jsjoeio
+  // Fix this once https://github.com/microsoft/playwright-test/issues/240
+  // is fixed
+  if (STORAGE) {
+    const storageState = JSON.parse(STORAGE) || {}
+    options.contextOptions = {
+      storageState,
+    }
+  }
+  test.beforeEach(async ({ page }) => {
+    codeServer = new CodeServer(page)
+    await codeServer.setup()
+    // NOTE@jsjoeio
+    // We're not using tmpdir from src/node/constants
+    // because Playwright doesn't fully support ES modules from
+    // the erorrs I'm seeing
+    tmpFolderPath = fs.mkdtempSync(path.join(tmpdir(), "code-server-test"))
```

Comment:
Definitely a nit but i'd love if we use the same path format as the other tests.

We may also need to create the parent directory since I'm not sure `mkdtemp` is recursive. 

I think we should stick to the async pattern to be consistent as well.

Should we make a `tmpdir` constant in the test utils for now since we can't import the other one?

```suggestion
    const tmpdir = path.join(tmpdir(), "code-server/tests")
    const testdir = path.join(tmpdir, "terminal")
    await fs.mkdir(testdir, { recursive: true }) // this could be in beforeAll so it's only done once
    
    tmpFolderPath = await fs.mkdtemp(path.join(testdir, "temp-"))
```

Reply:
yes! I'll refactor to use the suggestion for `tmp` in constants that you added above!